tests/run/test_patma.py

```diff
@@ -0,0 +1,3415 @@
+### COPIED FROM CPython 3.9
+### Original part after ############
+# cython: language_level=3
+
+# new code
+import cython
+from Cython.Compiler.Main import compile as cython_compile, CompileError
+from Cython.Build.Inline import cython_inline
+import contextlib
+from tempfile import NamedTemporaryFile
+
+@contextlib.contextmanager
+def hidden_stderr():
+    try:
+        from StringIO import StringIO
+    except ImportError:
+        from io import StringIO
+
+    old_stderr = sys.stderr
+    try:
+        sys.stderr = StringIO()
+        yield
+    finally:
+        sys.stderr = old_stderr
+
+def _compile(code):
+    with NamedTemporaryFile(suffix='.py') as f:
+        f.write(code.encode('utf8'))
+        f.flush()
+
+        with hidden_stderr():
+            result = cython_compile(f.name, language_level=3)
+    return result
```

Comment:
The tests only seem to look for syntax errors, so it's probably enough to use a `TreeFragment` here.

Reply:
Yes that seems to work so it's done now.